src/dotnet-interactive-vscode-common/src/notebookCellStatusBarItemProvider.ts

```diff
@@ -78,8 +78,9 @@ class DotNetNotebookCellStatusBarItemProvider {
             return [];
         }
 
+        const notebookMetadata = metadataUtilities.getNotebookDocumentMetadataFromNotebookDocument(cell.notebook);
         const cellMetadata = metadataUtilities.getNotebookCellMetadataFromNotebookCellElement(cell);
-        const cellKernelName = cellMetadata.kernelName ?? 'csharp';
+        const cellKernelName = cellMetadata.kernelName ?? notebookMetadata.kernelInfo.defaultKernelName;

```

Comment:
This is the core of the fix; everything else is just making sure we have something meaningful here.